### PR TITLE
Add `SHOW VSCHEMA KEYSPACES` query

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds a new Vitess specific query syntax to vtgates, `SHOW VSCHEMA KEYSPACES`. This query is intended to print out the information of all the keyspaces that Vitess has. Specifically, this query prints out the keyspace name, whether it is sharded, what the foreign key management mode is and the error in the keyspace (if any).

This query is intended to provide an easier way to look at the keyspace information as an alternative to looking at the `/debug/vschema` endpoint of vtgate and finding the required information. 

The query looks like the following when run from vtgate - 
```sql
mysql> show vschema keyspaces;
+---------------+---------+------------------+-------+
| Keyspace Name | Sharded | Foreign Key Mode | Error |
+---------------+---------+------------------+-------+
| uks           | false   | managed          |       |
| ks            | true    | managed          |       |
+---------------+---------+------------------+-------+
2 rows in set (0.00 sec)
```
## Related Issue(s)

- Came up as part of https://github.com/vitessio/vitess/issues/14495#issuecomment-1804855616

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by harshit-gangal — CHANGES_REQUESTED** [review]
need to fix upgrade downgrade test
```
2023-11-10T09:17:03.2520559Z === RUN   TestInitAndUpdate
2023-11-10T09:17:03.3682409Z     st_sharded_test.go:195: 
2023-11-10T09:17:03.3683627Z         	Error Trace:	/home/runner/work/vitess/vitess/go/test/endtoend/utils/utils.go:153
2023-11-10T09:17:03.3686117Z         	            				/home/runner/work/vitess/vitess/go/test/endtoend/utils/utils.go:48
2023-11-10T09:17:03.3689435Z         	            				/home/runner/work/vitess/vitess/go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:195
2023-11-10T09:17:03.3690912Z         	Error:      	Received unexpected error:
2023-11-10T09:17:03.3692982Z         	            	syntax error at position 23 near 'KEYSPACES' (errno 1105) (sqlstate HY000) during query: SHOW VSCHEMA KEYSPACES
2023-11-10T09:17:03.3694454Z         	Test:       	TestInitAndUpdate
2023-11-10T09:17:03.3695102Z         	Messages:   	for query: SHOW VSCHEMA KEYSPACES
2023-11-10T09:17:03.3695708Z --- FAIL: TestInitAndUpdate (0.12s)
```

File: changelog/19.0/19.0.0/summary.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
   - **[Deprecations and Deletions](#deprecations-and-deletions)**
   - **[Docker](#docker)**
     - [New MySQL Image](#mysql-image)
+  - **[VTGate](#vtgate)**
```
Comment (harshit-gangal) [inline changelog/19.0/19.0.0/summary.md:9]: Let's mark the header as Query Compatibility. 
We can add more query support in this section when added.

File: changelog/19.0/19.0.0/summary.md [hunk 2]
```diff
@@ -21,3 +23,23 @@ In `v19.0` the Vitess team is shipping a new image: `vitess/mysql`.
...
 
 Several tags are available to let you choose what version of MySQL you want to use: `vitess/mysql:8.0.30`, `vitess/mysql:8.0.34`.
+
+### <a id="vtgate"/>VTGate
+
+#### <a id="show-vschema-keyspaces"/>`SHOW VSCHEMA KEYSPACES` Query
+
+A SQL query, `SHOW VSCHEMA KEYSPACES` is now supported in Vitess. This query prints the vschema information
+for all the keyspaces. It is useful for seeing the foreign key mode, whether the keyspace is sharded, and if there is an
+error in the keyspace.
```
Comment (harshit-gangal) [inline changelog/19.0/19.0.0/summary.md:33]:
```suggestion
for all the keyspaces. It is useful for seeing the foreign key mode, whether the keyspace is sharded, and if there is an error in the VSchema for the keyspace.
```

File: go/vt/vtgate/planbuilder/show.go [hunk 3]
```diff
@@ -641,6 +644,26 @@ func buildEnginesPlan() (engine.Primitive, error) {
...
+		row = append(row, sqltypes.NewVarChar(fkMode.String()))
+		ksError := ""
+		if ks.Error != nil {
+			ksError = ks.Error.Error()
+		}
+		row = append(row, sqltypes.NewVarChar(ksError))
+		rows = append(rows, row)
+	}
+
+	return engine.NewRowsPrimitive(rows, buildVarCharFields("Keyspace Name", "Sharded", "Foreign Key Mode", "Error")), nil
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/show.go:664]:
```suggestion
	return engine.NewRowsPrimitive(rows, buildVarCharFields("Keyspace", "Sharded", "Foreign Key", "Comment")), nil
```

`Comment` will be more generic and can contain the error. Possibly can be used for additional information.

File: changelog/19.0/19.0.0/summary.md (hunk 1 above)
Comment (GuptaManan100) [reply changelog/19.0/19.0.0/summary.md]: Done

File: changelog/19.0/19.0.0/summary.md (hunk 2 above)
Comment (GuptaManan100) [reply changelog/19.0/19.0.0/summary.md]: Done

File: go/vt/vtgate/planbuilder/show.go (hunk 3 above)
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/show.go]: Done

**Review by harshit-gangal — APPROVED** [review]

**Review by systay — APPROVED** [review]

**Pull request merged**